src/UnoFlipModel.java

```diff
@@ -12,368 +12,228 @@
+        this.views = new ArrayList<UnoFlipView>(); //array list of views
     }
 
-    /**
-     * Gets the index of the player whose turn it is.
-     *
-     * @return The index of the current player.
-     */
-    public int getCurrentTurn() {
-        return currentTurn;
-    }
-
-    /**
-     * Sets the top card of the game.
-     *
-     * @param topCard The card to be set as the top card.
-     */
-    public void setTopCard(Card topCard) {
-        this.topCard = topCard;
-    }
-    /**
-     * Gets the current player.
-     *
-     * @return The current player.
-     */
-    public Player getCurrentPlayer(){
-        return players.get(currentTurn);
-    }
-    /**
-     * Flips the direction of the game.
-     */
-    public void flipTurnDirection(){
-        turnDirection = !turnDirection;
-    }
-    /**
-     * Gets the index of the next player.
-     *
-     * @return The index of the next player.
-     */
-    public int getNextTurn(){
-        return nextPlayerIndex;
-    }
-
-    /**
-     * Returns an ArrayList of players in the current game,used for testing only
-     * @return The ArrayList of players
-     */
-    public ArrayList<Player> getPlayers() {
-        return players;
-    }
-
-    /**
-     * Returns the turn direction,used for testing only
-     * @return The turn direction
-     */
-    public boolean getTurnDirection() {
-        return turnDirection;
-    }
-
-    /**
-     * Returns the number of players,used for testing only
-     * @return The number of players
-     */
-    public int getNumPlayers() {
-        return numPlayers;
-    }
-
-    /**
-     * Returns the string representation of current status
-     *
-     * @return string representation of current status
-     */
-    public String getStatus() {
-        return status;
-    }
-
-
-    /**
-     * Setting the status
-     * @param status - the new status
-     */
-    public void setStatus(String status ){
-        this.status = status;
-    }
-
-
-    /**
-     * Returns the chosen card index of the user,used for testing only
-     * @return The card index chosen by the user
-     */
-    public int getChosenCardIndex(){
-        return chosenCardIndex;
-    }
-
-    /**
-     * Returns the top card in play,used for testing only
-     * @return The top card
-     */
-    public Card getTopCard(){
-        return topCard;
-    }
 
     /**
      * Returns the number of players, used for testing only
-     * will only set the nunber of players if its between 2-4 players
-     * @param numPlayers The number of players
-     * @return the number of players that of the game, if invalid will return 0
+     * will only set the number of players if its between 2-4 players
+     * @param numPlayers The number of players received from the UnoFLipController
+     * @return the number of players that of the game
      */
     public int setNumPlayers(int numPlayers) {
-
-        if(!(numPlayers < 2 || numPlayers > 4)) {  //2-4 players
```

Comment:
Took it out for the same reason that you gave tharusha for taking out his try-catch. we are confident that the number received (from the set values of the combo box) are between 2-4 so there is no possible way for it to be a number we dont want. Either we have to keep both error checks for good practice or remove both since we will never get those errors.

Reply:
fair. i think now that i look at the model in isolation, it would be better to keep the error checking in with an exception